### DISCUSSION TITLE
Detect server restart with --watch ?

### DISCUSSION THREAD

**Discussion by alexgleason** [denoland/deno#25601] [Q&A]
I am trying to call `pglite.close()` before the process exits to avoid corrupting my database, but every time I edit my code with `--watch` it gets corrupted and I have to delete/recreate my database.

I used `Deno.addSignalListener('SIGINT', ...)` to clean up (and also added `SIGTERM`, `SIGHUP`, etc), but I can't figure out how to detect **when the server will restart due to a code change when using --watch**. I need to also call my cleanup function in that case. I think? I'm not completely sure if this is the issue or I'm running into a different bug.

Any ideas?

**Comment by bartlomieju** [comment]
Converted to an issue at https://github.com/denoland/deno/issues/25609, let's continue there.
